src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/XGBoostEvidenceFilter.java

```diff
@@ -0,0 +1,609 @@
+        final int numOverlap = evidenceFeatureCache[NUM_OVERLAP];
+        final int numCoherent = evidenceFeatureCache[NUM_COHERENT];
+        final int overlapMappingQuality = evidenceFeatureCache[OVERLAP_MAPPING_QUALITY];
+        final int coherentMappingQuality = evidenceFeatureCache[COHERENT_MAPPING_QUALITY];
+        final double meanOverlapMappingQuality = ((double)overlapMappingQuality) / numOverlap;
+        return new CoverageScaledOverlapInfo(numOverlap, numCoherent, overlapMappingQuality, coherentMappingQuality,
+                meanOverlapMappingQuality, coverage);
+    }
+
+    private CoverageScaledOverlapInfo getClusterOverlapInfo(final BreakpointEvidence evidence) {
+        int clusterNumOverlap = 0;
+        int clusterNumCoherent = 0;
+        int clusterOverlapMappingQuality = 0;
+        int clusterCoherentMappingQuality = 0;
+        double clusterMeanOverlapMappingQuality = 0.0;
+        for (final Iterator<BreakpointEvidence> overlapper_itr = evidenceOverlapChecker.overlappers(evidence); overlapper_itr.hasNext(); ) {
+            final BreakpointEvidence overlapper = overlapper_itr.next();
+            if (overlapper == evidence) {
+                continue; // don't count self-overlap in cluster features
+            }
+            if(!rawFeatureCache.containsKey(overlapper)) {
+                cacheOverlapInfo(overlapper);
+            }
+            final int[] overlapperFeatureCache = rawFeatureCache.get(overlapper);
+            clusterNumOverlap = Math.max(clusterNumOverlap, overlapperFeatureCache[NUM_OVERLAP]);
+            clusterNumCoherent = Math.max(clusterNumCoherent, overlapperFeatureCache[NUM_COHERENT]);
+            clusterOverlapMappingQuality = Math.max(clusterOverlapMappingQuality, overlapperFeatureCache[OVERLAP_MAPPING_QUALITY]);
+            clusterCoherentMappingQuality = Math.max(clusterCoherentMappingQuality, overlapperFeatureCache[COHERENT_MAPPING_QUALITY]);
+
+            final double meanOverlapMappingQuality = ((double) overlapperFeatureCache[OVERLAP_MAPPING_QUALITY]) / overlapperFeatureCache[NUM_OVERLAP];
+            clusterMeanOverlapMappingQuality = Math.max(clusterMeanOverlapMappingQuality, meanOverlapMappingQuality);
+        }
+
+        return new CoverageScaledOverlapInfo(clusterNumOverlap, clusterNumCoherent, clusterOverlapMappingQuality,
+                clusterCoherentMappingQuality, clusterMeanOverlapMappingQuality, coverage);
+    }
+
+    private static <T> void addToTree( final SVIntervalTree<List<T>> tree,
+                                       final SVInterval interval,
+                                       final T value ) {
+        final SVIntervalTree.Entry<List<T>> entry = tree.find(interval);
+        if ( entry != null ) {
+            entry.getValue().add(value);
+        } else {
+            final List<T> valueList = new ArrayList<>(1);
+            valueList.add(value);
+            tree.put(interval, valueList);
```

Comment:
`tree.put(interval, Collections.singletonList(value))`

Reply:
This list needs to be mutable since in principle multiple pieces of BreakpointEvidence can occupy the same location.